src/contracts/core/AVSDirectory.sol

```diff
@@ -383,6 +385,10 @@ contract AVSDirectory is
             // Mutate `isMember` to `true`.
             isMember[avs][operator][operatorSetIds[i]] = true;
 
+            unchecked {
```

Comment:
It's safe assuming the invariant that operatorSetMemberCount can only be non-zero if some isMember slot is non-zero. For deregistration, unchecked subtraction is safe because we assert that the operator is actually a member, and given the above invariant underflow is not possible. For registration, unchecked addition is fine because overflow is not reasonably possible (would require 2**256-2 operator sets).

Reply:
Yeah, that's what I'm trying to point out that this code depends on presently held assumptions versus having those assumptions proved by the specific code block.   I don't disagree with your reasoning, but many security bugs come about because closely held assumptions change over time.   That is what I meant by 'unsafe.'